axelor-supplychain/src/main/java/com/axelor/apps/supplychain/db/repo/ProductSupplychainRepository.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2018 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.supplychain.db.repo;
+
+import com.axelor.apps.stock.db.StockLocationLine;
+import com.axelor.apps.stock.db.repo.ProductStockRepository;
+import com.axelor.apps.stock.db.repo.StockLocationLineRepository;
+import com.google.inject.Inject;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+
+public class ProductSupplychainRepository extends ProductStockRepository {
+
+  @Inject private StockLocationLineRepository stockLocationLineRepo;
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  @Override
+  public Map<String, Object> populate(Map<String, Object> json, Map<String, Object> context) {
+    try {
+      Long productId = (Long) json.get("id");
+
+      if (context.get("_parent") != null) {
+        Map<String, Object> _parent = (Map<String, Object>) context.get("_parent");
+
+        Long stockLocationId = null;
+        if (context.get("_model").toString().equals("com.axelor.apps.stock.db.StockMoveLine")) {
+          if (_parent.get("fromStockLocation") != null) {
+            stockLocationId =
+                Long.parseLong(((Map) _parent.get("fromStockLocation")).get("id").toString());
+          }
+        } else {
+          if (_parent.get("stockLocation") != null) {
+            stockLocationId =
+                Long.parseLong(((Map) _parent.get("stockLocation")).get("id").toString());
+          }
+        }
+
+        if (stockLocationId != null) {
+          List<StockLocationLine> stockLocationLineList =
+              stockLocationLineRepo
+                  .all()
+                  .filter(
+                      "self.product.id = ?1 and self.stockLocation.id = ?2",
+                      productId,
+                      stockLocationId)
+                  .fetch();
+
```

Comment:
We should get only one stockLocationLine for same product and stockLocation, so here a fetchOne method is enougth.